mcs/class/Mono.Posix/Mono.Unix/UnixMarshal.cs

```diff
@@ -325,27 +325,34 @@ public static IntPtr StringToHeap (string s, int index, int count, Encoding enco
 			if (encoding == null)
 				throw new ArgumentNullException ("encoding");
 
-			int min_byte_count = encoding.GetMaxByteCount(1);
-			char[] copy = s.ToCharArray (index, count);
-			byte[] marshal = new byte [encoding.GetByteCount (copy) + min_byte_count];
+			if (index < 0 || count < 0)
```

Comment:
> That's done below.

*Where* "below"? This comment is in `UnixMarshal.cs`, not `UnixEncoding.cs`. There is no "below" in `UnixMarshal.cs`.